src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -255,5 +255,27 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
 
         var aggregator = _factory.Get(route);
         return aggregator.Aggregate(route, httpContext, contexts);
+    }
+
+    protected virtual async Task<Stream> CloneRequestBodyAsync(HttpRequest request, CancellationToken aborted)
+    {
+        request.EnableBuffering();

```

Comment:
### Next goal
- [EnableBuffering(HttpRequest, Int32, Int64)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httprequestrewindextensions.enablebuffering?view=aspnetcore-8.0#microsoft-aspnetcore-http-httprequestrewindextensions-enablebuffering(microsoft-aspnetcore-http-httprequest-system-int32-system-int64))

Let's add `MultiplexingMiddleware` options.

**Given** options for buffering from `FileAggregateRoute` class:
```csharp
public struct BufferingOptions
{
    public int BufferThreshold;
    public long BufferLimit;
}
```
**Then**, let's apply the options:
```suggestion
    protected virtual async Task<Stream> CloneRequestBodyAsync(HttpRequest request, CancellationToken aborted, BufferingOptions options)
    {
        request.EnableBuffering(options.BufferThreshold, options.BufferLimit);
```

Reply:
Your [concerns](https://github.com/ThreeMammals/Ocelot/pull/2050#discussion_r1567333939) about introducing this feature have been noted. It is designed to be non-global and solely for multiplexing, as buffering is permitted only for multiplexed requests.

@ggnaegi, could you share your thoughts on this?